project/threads/api.py

```diff
@@ -1,16 +1,520 @@
+
+    location = (
+        req_edit_thread.level
+        if not req_edit_thread.state
+        else dict(US_STATES).get(req_edit_thread.state)
+    )
+
+    return_data = {
+        "thread_id": thread_id,
+        "title": req_edit_thread.title,
+        "summary": req_edit_thread.summary,
+        "category": {
+            "id": req_edit_thread.category.id,
+            "name": req_edit_thread.category.name,
+        },
+        "level": req_edit_thread.level,
+        "state": req_edit_thread.state if req_edit_thread.level == "state" else "",
+        "location": location,
+    }
+    return JsonResponse({"data": return_data})
+
+
+@login_required
+def edit_user(request):
+    """
+    Edit Profile Model
+    """
+    request_data = request.POST
+    user = request.user
+    account = Profile.objects.get(user=user)
+
+    data = {
+        "first_name": request_data.get("first_name", account.first_name),
+        "last_name": request_data.get("last_name", account.last_name),
+        "about_me": request_data.get("about_me", account.about_me),
+    }
+
+    account.__dict__.update(data)
+
+    try:
+        account.save()
+    except Exception as e:
+        # print('EXCEPTION THROWN HERE!! ')
+        return HttpResponseServerError(reason=str(e))
+
+        account.refresh_from_db()
+
+    return JsonResponse(Profile.objects.summarize(account))
+
+
+class UpdateProfileImage(forms.ModelForm):
+    """
+    Form for updating profile image
+    """
+
+    class Meta:
+        model = Profile
+        fields = ["profile_image"]
+
+    profile_image = forms.ImageField()
+
+    def clean_profile_image(self):
+        """
+        This function is used to make sure that profile images follow Civiwiki standards.
+
+        Requirements:
+            - Height cannot exceed 960px
+            - Width cannot exceed 1280px
+            - Image must be (jpg, jpeg, pjeg, png)
+            - File size cannot exceed 2MB
+        """
+        profile_image = self.cleaned_data["profile_image"]
+
+        try:
+            w, h = get_image_dimensions(profile_image)
+
+            # validate dimensions
+            max_height = 960
+            max_width = 1280
+            if w > max_width or h > max_height:
+                raise forms.ValidationError(
+                    u"Please use an image that is {w} x {h} pixels or smaller.".format(
+                        w=max_width, h=max_height
+                    )
+                )
+
+            # validate content type
+            main, sub = profile_image.content_type.split("/")
+            if not (main == "image" and sub in ["jpg", "jpeg", "pjpeg", "png"]):
+                raise forms.ValidationError(u"Please use a JPEG or PNG image.")
+
+            # validate file size
+            if len(profile_image) > (2000 * 1024):
+                raise forms.ValidationError(
+                    "Profile image file size may not exceed 2MB."
+                )

```

Comment:
This code is related to the `accounts` app and should not be moved as part of this pull request. We want to only move the thread-related code mentioned in #1032 as part of this pull request.
```suggestion
```